src/ipc/Messages.h

```diff
@@ -17,16 +17,23 @@ namespace Ipc
 {
 
 /// message class identifier
-typedef enum { mtNone = 0, mtRegistration,
-               mtStrandSearchRequest, mtStrandSearchResponse,
-               mtSharedListenRequest, mtSharedListenResponse,
+typedef enum { enumBegin_ = 0,
+               mtRegistration, ///< strand registration with Coordinator (also used as an ACK)
```

Comment:
There are two WholeEnum API-related problems here:

1. Per WholeEnum API, `enumBegin_` must match the first "real" value but currently does not. To solve this problem, we cannot just say `mtRegistration = enumBegin_` because we (correctly) use the zero value as an invalid/impossible one. AFAICT, the correct solution requires restoring mtNone.

2. The enumBegin_/enumEnd_ names are only allowed in scoped [enumeration](https://en.cppreference.com/w/cpp/language/enum) -- same names from different _unscoped_ enums would clash. I am not sure we should switch MessageType to scoped enumeration, but if we do want to switch, we should do it in _this_ PR. It would be nice to come up with some rules for when scoped enumeration is preferred. Are there any existing general (i.e. not Squid-specific) rules for that?

If we do not switch to scoped enums:

```C++
typedef enum {
               mtNone = 0, ///< unspecified or unknown message kind; unused on the wire
               mtRegistration, ///< strand registration with Coordinator (also used as an ACK)
...
               mtEnd ///< for message kind range checks; unused on the wire
             } MessageType;
```

Reply:
Since currently MessageType is not used with WholeEnum I think we should not bother switching to the scoped enum. Done with mtNote/mtEnd.

> two WholeEnum API-related problems 

Note that some existing Squid enums have these problems (e.g, icp_opcode or Http::HdrType).